demos/demo35_FD_gradient.py

```diff
@@ -0,0 +1,27 @@
+# %% Import required packages
+import cuqi
+import matplotlib.pyplot as plt
+
+#%% Create test problem
+TP = cuqi.testproblem.Heat_1D(dim=60, max_time=0.01, field_type='Step')
+
+#%% Get the posterior
+posterior = TP.posterior
+
+#%% Sample from the posterior using Metropolis-Hastings
+MH_sampler = cuqi.sampler.MetropolisHastings(posterior)
+MH_samples = MH_sampler.sample_adapt(1000)
+MH_samples.plot_ci(95,exact=TP.exactSolution)
+
+#%% Sample from the posterior using MALA
+plt.figure()
+posterior.enable_FD()
```

Comment:
Would it be illustrative to make an attempt to call MALA without having enabled FD and seeing it fail? The script would not run, so another idea would be to put the attempt in a commented code line, and explain that had it been run it would have failed, because it requires gradients which we don't have off the bat, so need to enable the FD.

Reply:
This is really a good point. I tried to capture it with some print statement and try, except clause.